oci/archive/oci_transport.go

```diff
@@ -69,11 +71,27 @@ func NewReference(file, image string) (types.ImageReference, error) {
 		return nil, err
 	}
 
-	if err := internal.ValidateImageName(image); err != nil {
-		return nil, err
+	var img string
+	index := -1
+	if strings.HasPrefix(image, "@") {
```

Comment:
The idea of `NewReference` is that the inputs are already well-typed; it shouldn’t be doing string parsing.

Please move the string parsing into ParseReference (and then further mostly into the `internal` subpackage), and add a public a constructor that allows creating the index references from raw data.

---

It would also be nice for the oci/archive and oci/layout reference API, and a fair bit of the implementation, to remain the same (or shared in `internal`), although the two will necessarily diverge a bit with the support for long-term readers/writers.